lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
@tgamblin is there a flags section already?  or I should add it?  These flags are required to build test programs during the cmake configuration phase and that is before cmake can add flags in the generated makefiles.  So these flags needs to be passed to the compiler by default.